kaldb/src/test/java/com/slack/kaldb/clusterManager/RecoveryTaskAssignmentServiceTest.java

```diff
@@ -0,0 +1,724 @@
+
+import static com.slack.kaldb.config.KaldbConfig.DEFAULT_START_STOP_DURATION;
+import static org.assertj.core.api.AssertionsForClassTypes.assertThat;
+import static org.awaitility.Awaitility.await;
+import static org.junit.Assert.assertTrue;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.Mockito.doCallRealMethod;
+import static org.mockito.Mockito.spy;
+
+import brave.Tracing;
+import com.google.common.util.concurrent.Futures;
+import com.slack.kaldb.metadata.recovery.RecoveryNodeMetadata;
+import com.slack.kaldb.metadata.recovery.RecoveryNodeMetadataStore;
+import com.slack.kaldb.metadata.recovery.RecoveryTaskMetadata;
+import com.slack.kaldb.metadata.recovery.RecoveryTaskMetadataStore;
+import com.slack.kaldb.metadata.zookeeper.MetadataStore;
+import com.slack.kaldb.metadata.zookeeper.ZookeeperMetadataStoreImpl;
+import com.slack.kaldb.proto.config.KaldbConfigs;
+import com.slack.kaldb.proto.metadata.Metadata;
+import com.slack.kaldb.testlib.MetricsUtil;
+import io.micrometer.core.instrument.MeterRegistry;
+import io.micrometer.core.instrument.simple.SimpleMeterRegistry;
+import java.io.IOException;
+import java.time.Instant;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.UUID;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+import org.apache.curator.test.TestingServer;
+import org.assertj.core.api.Assertions;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+
+public class RecoveryTaskAssignmentServiceTest {
+
+  private TestingServer testingServer;
+  private MeterRegistry meterRegistry;
+
+  private MetadataStore metadataStore;
+  private RecoveryTaskMetadataStore recoveryTaskMetadataStore;
+  private RecoveryNodeMetadataStore recoveryNodeMetadataStore;
+
+  @Before
+  public void setup() throws Exception {
+    Tracing.newBuilder().build();
+    meterRegistry = new SimpleMeterRegistry();
+    testingServer = new TestingServer();
+
+    KaldbConfigs.ZookeeperConfig zkConfig =
+        KaldbConfigs.ZookeeperConfig.newBuilder()
+            .setZkConnectString(testingServer.getConnectString())
+            .setZkPathPrefix("RecoveryTaskAssignmentServiceTest")
+            .setZkSessionTimeoutMs(1000)
+            .setZkConnectionTimeoutMs(1000)
+            .setSleepBetweenRetriesMs(1000)
+            .build();
+
+    metadataStore = ZookeeperMetadataStoreImpl.fromConfig(meterRegistry, zkConfig);
+    recoveryTaskMetadataStore = spy(new RecoveryTaskMetadataStore(metadataStore, true));
+    recoveryNodeMetadataStore = spy(new RecoveryNodeMetadataStore(metadataStore, true));
+  }
+
+  @After
+  public void shutdown() throws IOException {
+    recoveryNodeMetadataStore.close();
+    recoveryTaskMetadataStore.close();
+    metadataStore.close();
+
+    testingServer.close();
+    meterRegistry.close();
+  }
+
+  @Test(expected = IllegalArgumentException.class)
+  public void shouldCheckInvalidConfigs() {
```

Comment:
Added an additional comment expounding on the one in the `RecoveryAssignmentService` that explains why we neither check nor test the scheduled service params.

Reply:
We should still add tests for the other values since these are application invariants. For example, if we change the underlying implementation that doesn't check for negative values, without these unit tests we would introduce a bug.

Reply:
Added additional tests for invalid schedule configurations, updated comment.